rust/lance/src/index.rs

```diff
@@ -1873,6 +1890,92 @@ mod tests {
             .is_err());
     }
 
+    #[tokio::test]
+    async fn test_bitmap_index_statistics_minimal_io_via_dataset() {
+        const NUM_ROWS: usize = 500_000;
+        let test_dir = TempStrDir::default();
+        let schema = Arc::new(Schema::new(vec![Field::new(
+            "status",
+            DataType::Int32,
+            false,
+        )]));
+        let values: Vec<i32> = (0..NUM_ROWS as i32).collect();
+        let batch =
+            RecordBatch::try_new(schema.clone(), vec![Arc::new(Int32Array::from(values))]).unwrap();
+        let reader = RecordBatchIterator::new(vec![Ok(batch)], schema.clone());
+
+        let io_tracker = Arc::new(IOTracker::default());
+        let write_params = WriteParams {
+            store_params: Some(ObjectStoreParams {
+                object_store_wrapper: Some(io_tracker.clone()),
+                ..Default::default()
+            }),
+            ..WriteParams::default()
+        };
+
+        Dataset::write(reader, &test_dir, Some(write_params))
+            .await
+            .unwrap();
+
+        let read_params = ReadParams {
+            store_options: Some(ObjectStoreParams {
+                object_store_wrapper: Some(io_tracker.clone()),
+                ..Default::default()
+            }),
+            ..ReadParams::default()
+        };
+
+        let mut dataset = DatasetBuilder::from_uri(&test_dir)
+            .with_read_params(read_params)
+            .load()
+            .await
+            .unwrap();
```

Comment:
I recently made it so you get an IO tracker on every object store by default, so you can simplify this to:
```suggestion
        let mut dataset = Dataset::write(reader, &test_dir, Some(write_params))
            .await
            .unwrap();
        let io_tracker = dataset.object_store().io_tracker().clone();
```